### ISSUE TITLE
Truffle & OpenZeppelin tutorial issues

### ISSUE THREAD

**Issue by medied** [trufflesuite/trufflesuite.github.io#190]
I'm running into issues interacting with `TutorialToken` from the [OpenZeppelin](http://truffleframework.com/tutorials/robust-smart-contracts-with-openzeppelin) tutorial (section 4 to be precise). Behavior I'm getting and other info:

* Balance field in the front end shows up as empty
* Transfer option seems to be unresponsive
* Not sure what I'm missing with Metamask exactly. I'm connected to http://localhost:8545/ and `testrpc` seems to be working fine. Tried adding the token as [this doc suggests](https://support.metamask.io/kb/article/4-managing-tokens).
* I [asked in Gitter](https://gitter.im/ConsenSys/truffle?at=59b99814177fb9fe7eb91ae6) before creating this issue. 
* What it looks like:

![screen shot 2017-09-13 at 5 27 05 pm](https://user-images.githubusercontent.com/4752599/30401709-36f6391e-98a9-11e7-9132-c565da5fe8c4.png)

Some questions and thoughts: 
* Should Metamask be connected to http://localhost:3000/ where the app is running? This doesn't seem to be the case.
* How exactly should the app interact with Metamask? Could someone expand on the part below from the tutorial?

>Our basic dapp shows the TutorialToken balance of the selected account in MetaMask.
>
>Try transferring some tokens to a different account, here we moved 2000 TT to the second account.

Any pointers would be helpful as I'm trying to learn as much as possible. Thanks in advance!

**Comment by DiscRiskandBisque** [comment]
Here are some things to try, there could be a few different causes of this.

- MetaMask should be connected to Localhost 8545. To be sure, click the network dropdown next to the fox head in the popup window.
- MetaMask must be unlocked. If you're getting the error "Invalid number of arguments to Solidity function", this is the cause.
- To connect MetaMask to the particular testrpc instance you created when migrating the contracts, you'll need to take a few steps. 1. Copy the mnemonic phrase from when the testrpc first starts. If your console won't let you scroll up far enough to see it anymore, restart the rpc, remigrate and copy it. 2. If you're past the MetaMask unlock screen, click the three-bar icon in the upper-right and select Lock. 3. Click the Restore from seed phrase link. 4. Paste in the mnemonic you copied from the testrpc and set any password you can easily remember. An account with a little less than 100 ETH should appear (or exactly 100 if you haven't migrated yet since starting the testrpc)

If all the above fails, let's see if it works without MetaMask:

- Capitalize web3 on line 16 of src/js/app.js, the new line should read: `App.web3Provider = new Web3.providers.HttpProvider('http://localhost:8545');`
- Open a new incognito window (assuming you haven't given MM permission to run incognito) and try navigating to localhost:3000

**Comment by medied** [comment]
Thanks for reaching out @DiscRiskandBisque, here's what I found:

* App is working in incognito with the capital `Web3`, not working in incognito with the lowercase `web3`
* If I enable Metamask in incognito, it breaks and I see the same behavior as before.
* After doing a few transactions with the `Transfer` option in the app, I see:
** Metamask ETH balance not changing (assuming that's expected); [added the token](https://support.metamask.io/kb/article/4-managing-tokens) and do see that changing and matching app balance (also as I expect)
** I also do see a transaction record in the `testrpc` log
* I forgot to mention the console output earlier (saw [your Gitter message](https://gitter.im/ConsenSys/truffle?at=59b9ad0bcfeed2eb6507052f) 👍 ), when not working in non-incognito it outputs: `Adoption has not been deployed to detected network (network/artifact mismatch)` pointing to line 99 in `app.js`

In summary it works when I interact with the app in incognito, and Metamask in non-incognito. Curious if you have any other thoughts or reactions. I think this works for me for the time being, feel free to close this issue. Thanks again!

**Comment by adamclark64** [comment]
Thank you @medied. Spun my wheels for a bit until my first attempt at an incognito tab worked first try.

**Comment by bmmpxf** [comment]
Hi all. We've done a bunch of updates to this tutorial with this in mind. Please give it another try, as these issues should be resolved. You should not need an incognito window to perform the tutorial. But if you're still having issues, please let us know!

**Comment by bmmpxf** [comment]
Thanks @mjoecohen . Please use the https://github.com/trufflesuite/trufflesuite.com/issues repo for all tutorial issues from here on out. (We've migrated away from this repo.)
